src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,234 @@
+            //                   ↓                  | normalizedToken (dot)
+            // white?.Black.White.Black?.White      | current user input
+            // white?.Black.White.Black?.White      | rootExpression (text manipulation starts with this)
+            //       .Black.White                   | parentExpression (needed to calculate the position to insert the closing brace)
+            //                    Black             | identifier at cursor position (gets removed, because the user typed the name of a type)
+            // |----------------------|             | part to replace (TextChange.Span), if identifier is not present: ends at rootExpression.End (after White.)
+            //                   ↑                  | insert closing brace between White and dot (parentExpression.Span.End)
+            // ((Black)white?.Black.White).?.White  | The result. Because we removed the identifier, the remainder after the identifier may be syntactically wrong 
+            //                             ↑        | cursor after the manipulation is placed after the dot
+            var rootExpression = GetRootExpressionOfInvocation(normalizedToken);
+            var parentExpression = GetParentExpressionOfInvocation(normalizedToken);
+            var identifier = tokenAtPosition.Parent as IdentifierNameSyntax;
+            if (rootExpression is null || parentExpression is null)
+            {
+                return null;
+            }
+
+            var spanToReplace = TextSpan.FromBounds(rootExpression.Span.Start, identifier is null ? rootExpression.Span.End : identifier.Span.End);
+            var cursorPositionOffset = spanToReplace.End - position;
+            var fromRootToParent = rootExpression.ToString();
+            if (identifier is { })
+            {
+                // Cut of the identifier
+                var length = identifier.Span.Start - rootExpression.SpanStart;
+                fromRootToParent = fromRootToParent.Substring(0, length);
+                // place cursor right behind ).
+                cursorPositionOffset = 0;
+            }
+            var fromRootToParentWithInsertedClosingBracket = fromRootToParent.Insert(parentExpression.Span.End - rootExpression.SpanStart, ")");
+            var conversion = $"(({typeName}){fromRootToParentWithInsertedClosingBracket}";
+            var newPosition = spanToReplace.Start + conversion.Length - cursorPositionOffset;
+            return CompletionChange.Create(new TextChange(spanToReplace, conversion), newPosition);
+        }
+
+        private static async Task<CompletionChange?> HandleIndexerChangeAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            var token = root.FindTokenOnLeftOfPosition(position);
```

Comment:
why no `getpreviousiftouchingword`?  what if the user types `foo.this`

Reply:
I changed the filtertext for indexers from "[" to "this" (without the brace). The brace was not working anyway. I also added some tests for "this". See f1a17bbd